src/engine/init.c

```diff
@@ -283,9 +282,12 @@ dss_tgt_nr_get(unsigned int ncores, unsigned int nr, bool oversubscribe)
 		goto retry;
 	}
 
-	if (tgt_nr != nr)
+	if (tgt_nr != nr) {
+		if (!oversubscribe)
```

Comment:
This check is redundant, actually it should be "assert(!oversubscribe)", right?

Reply:
No, this is needed to return errors if oversubscribe is not enabled.